orchestrator/cli/generator/generator/migration.py

```diff
@@ -10,28 +10,67 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import itertools
 import re
 import subprocess  # noqa: S404
-from typing import Generator, Optional
+from datetime import datetime
+from os import environ
+from pathlib import Path
+from typing import Any, Callable, Dict, Generator, List, Optional, Tuple
 
 import structlog
+from alembic.util import rev_id  # type: ignore[attr-defined]
+from jinja2 import Environment
 
-from orchestrator.cli.generator.generator.helpers import get_variable
+from orchestrator.cli.generator.generator.helpers import product_types_module
+from orchestrator.cli.generator.generator.settings import product_generator_settings as settings
 
 logger = structlog.getLogger(__name__)
 
 
-def create_migration_file(config: dict) -> Optional[str]:
-    create_migration_command = f'PYTHONPATH=. DATABASE_URI=postgresql://nwa:nwa@localhost/nwa-workflows python main.py db revision --message "Add {config["name"]}" --head=data@head'
+def create_migration_file(message: str, head: str) -> Optional[Path]:
+    if not environ.get("DATABASE_URI"):
+        environ.update({"DATABASE_URI": "postgresql://nwa:nwa@localhost/orchestrator-core"})
+    if not environ.get("PYTHONPATH"):
+        environ.update({"PYTHONPATH": "."})
+    logger.info(
+        "creating new db revision", database_uri=environ.get("DATABASE_URI"), pythonpath=environ.get("PYTHONPATH")
+    )
+    create_migration_command = f'python main.py db revision --message "{message}" --head={head}@head'
     result = subprocess.check_output(create_migration_command, shell=True, text=True)  # noqa: S602
 
     for line in result.splitlines():
-        if m := re.search("migrations/versions/general/([^ ]+)", line):
-            return m[1]
+        if m := re.search("migrations/versions/schema/([^ ]+)", line):
+            return Path("migrations/versions/schema") / m[1]
     else:
         return None
 
 
+def get_heads() -> dict:
+    get_heads_command = "python main.py db heads"
+    result = subprocess.check_output(get_heads_command, shell=True, text=True)  # noqa: S602
+
+    heads = {}
+    for line in result.splitlines():
```

Comment:
Can probably be rewritten a bit nicer with set comprehension:

Semi working code:

```python
def is_revision(line):
   if match := re.search("^([^ ]+) \\(([^ ]+)\\)", line):
       return match.groups()
   return None

revisions = (revision for line in result.splitlines() if revision := is_revision(line))
return dict(revisions)
```